osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -487,6 +487,15 @@ public HitWindows FirstAvailableHitWindows
 
         protected virtual ResumeOverlay CreateResumeOverlay() => null;
 
+        /// <summary>
+        /// Whether to display the HUD with this ruleset.
+        /// Override to false to completely disable the display of the HUD with this ruleset.
+        /// </summary>
+        /// <remarks>
+        /// HUD refers here to <see cref="HUDOverlay"/> in player as well as <see cref="BreakOverlay"/>.
+        /// </remarks>
+        public virtual bool DisplayHud => true;
```

Comment:
any reason you didn't use the suggested naming? you also got the case incorrect for HUD. it's an acronym. and you can't just arbitrarily add other hidden elements and call it HUD because that acronym is already reserved.

Rename this to `AllowGameplayOverlays`